sycl/include/sycl/half_type.hpp

```diff
@@ -265,7 +266,11 @@ using Vec8StorageT = half_vec<8>;
 using Vec16StorageT = half_vec<16>;
 #endif
 
+#ifndef __SYCL_DEVICE_ONLY__
 class half {
+#else
+class [[__sycl_detail__::__uses_aspects__(aspect::fp16)]] half {
```

Comment:
The attribute is marked as `SilentlyIgnoreSYCLIsHost`, so I don't think we necessarily need the `ifndef` here. That said, I know for a number of other attributes with the same marker we do it, so it's just a non-blocking nit.

Reply:
It otherwise fails `-Werror` build of the runtime itself: [link](https://github.com/intel/llvm/actions/runs/3181849795/jobs/5187097459#step:7:8768)

Reply:
Ah, I suppose even if it didn't fail for DPC++ it would cause warnings when using other host compilers. Disregard my comment!